controllers/api/v1/chat_api.js

```diff
@@ -0,0 +1,367 @@
+const Chat = require('../../../models/chat');
+const ChatRoom = require('../../../models/chatRoom');
+
+module.exports.createChatRoom = async function (req, res) {
+    try {
+        const data = req.params;
+        console.log(data);
+        let chatRoomId = data.sender + data.receiver;
+        let reverseChatRoomId = data.receiver + data.sender;
+
+        if (data.type == 'global') {
```

Comment:
move this logic to saprate function. Move string to variable. Make it group and private